runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java

```diff
@@ -103,121 +97,118 @@
-	public void onConfigChanged(ConfigChanged event)
-	{
+	public void onConfigChanged(ConfigChanged event) {
 		abyssOverlay.updateConfig();
 	}
 
 	@Subscribe
-	public void onChatMessage(ChatMessage event)
-	{
-		if (event.getType() != ChatMessageType.GAMEMESSAGE)
-		{
+	public void onChatMessage(ChatMessage event) {
+		if (event.getType() != ChatMessageType.GAMEMESSAGE) {
 			return;
 		}
 
-		if (config.degradingNotification())
-		{
-			if (event.getMessage().contains(POUCH_DECAYED_MESSAGE))
-			{
+		if (config.degradingNotification()) {
+			if (event.getMessage().contains(POUCH_DECAYED_MESSAGE)) {
 				notifier.notify(POUCH_DECAYED_NOTIFICATION_MESSAGE);
 			}
 		}
 	}
 
 	@Subscribe
-	public void onMenuEntryAdded(MenuEntryAdded entry)
-	{
-		if (!wearingTiara)
-		{
+	public void onMenuEntryAdded(MenuEntryAdded entry) {
+		if (client.getGameState() != GameState.LOGGED_IN) {
+			return;
+		}
+
+		Widget loginScreenOne = client.getWidget(WidgetInfo.LOGIN_CLICK_TO_PLAY_SCREEN);
+		Widget loginScreenTwo = client.getWidget(WidgetInfo.LOGIN_CLICK_TO_PLAY_SCREEN_MESSAGE_OF_THE_DAY);
+
+		if (loginScreenOne != null || loginScreenTwo != null) {
 			return;
 		}
 
 		final String option = Text.removeTags(entry.getOption()).toLowerCase();
 		final String target = Text.removeTags(entry.getTarget()).toLowerCase();
-		final int id = entry.getIdentifier();
-		final int type = entry.getType();
 
-		if (target.contains("pouch") && !target.startsWith("rune"))
-		{
-			if (option.equals("deposit-all") && type == 57 && id == 2)
-			{
-				swap(client, "fill", option, target);
-				swap(client, "cancel", option, "", target);
-			}
-			else  if (option.equals("fill") && id != 9)
-			{
-				swap(client, "empty", option, target);
-			}
-		}
 
-		else if (target.contains("ring of dueling") && option.contains("remove"))
-		{
-			if (target.contains("7") || target.contains("5") || target.contains("3") || target.contains("1"))
-			{
-				swap(client, "castle wars", option, target);
+		Widget widgetBankTitleBar = client.getWidget(WidgetInfo.BANK_TITLE_BAR);
+		if (isEssencePouch(target)) {
+			if (widgetBankTitleBar == null || widgetBankTitleBar.isHidden()) { //swap pouches based on empty/full and in bank
+				swap(client, "Empty", option, target);
+			} else {
+				swap(client, "Fill", option, target); //Due to RuneLite issues the "Deposit" menutext will always show even though it is on fill
 			}
-			else if (wearingBindingNeck())
-			{
+		} else if (target.contains("ring of dueling") && option.contains("withdraw")) { //withdraw-1 ring of dueling
+			swap(client, "withdraw-1", option, target);
+		} else if (target.contains("binding necklace") && option.contains("withdraw")) { //withdraw-1 binding necklace
+			swap(client, "withdraw-1", option, target);
+		}
+		else if (target.contains("ring of dueling") && option.contains("remove")) {
+			if (client.getLocalPlayer().getWorldLocation().getRegionID() != 10315) { //changes duel ring teleport options based on location
```

Comment:
instead of checking if you're at the fire altar, check if you're at castle wars?

Reply:
People can bank at other places, I bank at crafting guild

Reply:
Can we get this swap for Stamina (1) and Binding necklaces :)